src/systems/gimbal_controller/GimbalControllerPlugin.cc

```diff
@@ -0,0 +1,484 @@
+		if(this->dataPtr->model.JointByName(_ecm, pitchJointName))
+		{
+			this->dataPtr->pitchJoint = 
+      this->dataPtr->model.JointByName(_ecm, "picthJointName");
+		}
+
+		else
+		{
+			ignwarn << "joint_pitch ["<< pitchJointName <<"] does not exist?\n"<< std::endl;
+		}
+	}
+
+	if(!this->dataPtr->pitchJoint)
+	{
+		ignerr << "GimbalControllerPlugin::Configure ERROR! Can't get pitch joint." 
+           <<pitchJointName <<"' " <<std::endl;
+	}
+
+	// get imu sensor
+	this->dataPtr->imuName = _sdf->Get("imuName", 
+    static_cast<std::string>("imu_sensor")).first;
+}
+
+
+void GimbalControllerPluginPrivate::Init(const EntityComponentManager &_ecm)
+{    
+  // receive pitch command via ignition transport 
+	std::string pitchTopic = this->model.Name(_ecm) + "/gimbal_pitch_cmd";
+  this->node.Subscribe(pitchTopic, 
+    &GimbalControllerPluginPrivate::OnPitchStringMsg);
+
+  // receive roll command via gz transport
+  std::string rollTopic = this->model.Name(_ecm) + "/gimbal_roll_cmd";
+  this->node.Subscribe(rollTopic, 
+    &GimbalControllerPluginPrivate::OnRollStringMsg);
+
+  // receive yaw command via gz transport
+  std::string yawTopic = this->model.Name(_ecm) + "/gimbal_yaw_cmd";
+  this->node.Subscribe(yawTopic, 
+    &GimbalControllerPluginPrivate::OnYawStringMsg);
+
+  // publish pitch status via gz transport
+  pitchTopic = this->model.Name(_ecm) + "/gimbal_pitch_status";
+  this->pitchPub= this->node.Advertise<ignition::msgs::StringMsg>(pitchTopic);
+
+  // publish roll status via gz transport
+  rollTopic = this->model.Name(_ecm) + "/gimbal_roll_status";
+  this->rollPub= this->node.Advertise<ignition::msgs::StringMsg>(rollTopic);
+
+  // publish yaw status via gz transport
+  yawTopic = this->model.Name(_ecm) + "/gimbal_yaw_status";
+  this->yawPub= this->node.Advertise<ignition::msgs::StringMsg>(yawTopic);
+
+  ignmsg<< "GimbalControllerPluginPrivate::Init" <<std::endl;
+}
+
+//////////////////////////////////////////////////
+void GimbalControllerPluginPrivate::OnPitchStringMsg
+(const msgs::StringMsg &_msg)
+{
+	this->pitchCommand = atof(_msg.data().c_str());
+}
+
+//////////////////////////////////////////////////
+void GimbalControllerPluginPrivate::OnYawStringMsg
+(const msgs::StringMsg &_msg)
+{
+	this->yawCommand = atof(_msg.data().c_str());
+}
+
+//////////////////////////////////////////////////
+void GimbalControllerPluginPrivate::OnRollStringMsg
+(const msgs::StringMsg &_msg)
+{
+	this->rollCommand = atof(_msg.data().c_str());
+}
+
+//////////////////////////////////////////////////
+ignition::math::Vector3d GimbalControllerPluginPrivate::ThreeAxisRot(
+  double r11, double r12, double r21, double r31, double r32)
+{
+  return ignition::math::Vector3d(
+    atan2( r31, r32),
+    asin ( r21 ),
+    atan2( r11, r12));
+}
+
+//////////////////////////////////////////////////
+ignition::math::Vector3d GimbalControllerPluginPrivate::QtoZXY
+(const ignition::math::Quaterniond &_q)
```

Comment:
This function isn't used, can it be removed?

Please go over all functions and variables and remove the ones which aren't necessary.